app/src/main/AndroidManifest.xml

```diff
@@ -142,7 +142,7 @@
 
         <provider
             android:permission="io.rapidpro.androidchannel.SERVICE_PERMISSION"
-            android:exported="false"
+            android:exported="true"
```

Comment:
The provider needs to be exported and we rely on the permission above to control who can access it

Reply:
I think you can make it so only things signed with our signature can access an exported service.